org.eclipse.winery.bpmn4tosca.converter.tobpel/src/main/java/org/eclipse/winery/bpmn2bpel/model/Gateway.java

```diff
@@ -8,9 +8,35 @@
  *
  * Contributors:
  *     Sebastian Wagner - initial API and implementation
+ *     ZTE - support of more gateways
+ *******************************************************************************/
+/*******************************************************************************
+ * Modifications Copyright 2017 ZTE Corporation.
  *******************************************************************************/
 package org.eclipse.winery.bpmn2bpel.model;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public abstract class Gateway extends Node {
+    private List<GatewayBranch> branchList = new ArrayList<GatewayBranch>();
+
+    public List<GatewayBranch> getBranchList() {
+        return branchList;
+    }
+
+    public void setBranchList(List<GatewayBranch> branches) {
+        this.branchList = branches;
+    }
+
+    public GatewayBranch getBranch(String id) {
```

Comment:
Would it be possible to switch to Java8's Optionals to avoid `null` objects?

`return Optional<GatewayBranch>`

Later in the code: `if (branch.isPresent())`